### PR TITLE
RavenDB-5773 Implement LazySessionOperations - WIP

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.NewClient/Commands/QueryOperation.cs [hunk 1]
```diff
@@ -146,9 +146,7 @@ public IList<T> Complete<T>()
                 string id;
                 if (document.TryGet(Constants.Metadata.Key, out metadata) == false)
                     throw new InvalidOperationException("Document must have a metadata");
-                if (metadata.TryGet(Constants.Metadata.Id, out id) == false)
-                    throw new InvalidOperationException("Document must have an id");
-
+                metadata.TryGet(Constants.Metadata.Id, out id);
```
Comment (ayende) [inline src/Raven.NewClient/Commands/QueryOperation.cs:149]: Why did you remove the check for missing id?

Comment (efratshenhar) [reply src/Raven.NewClient/Commands/QueryOperation.cs:149]: Because we handle  missing id in TrackEntity
